### PR TITLE
Add API for Disk Resize

### PR DESCRIPTION
**Pull request by baude**
Add disk resize api for pkg/hypvctl that takes a disk path and a strongly typed new size.  No error protection for invalid disk resizes (like trying to descrease the size of a disk).

Fixes: #54

### CODE REVIEW

File: pkg/hypervctl/disk_resize.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"os/exec"
+
+	"github.com/containers/podman/v4/pkg/strongunits"
+)
+
+// ResizeDisk takes a diskPath and strongly typed new size and uses powershell
+// to change its size.  There is no error protection for trying to size a disk
+// smaller than the current size.
+func ResizeDisk(diskPath string, newSize strongunits.GiB) error {
+	resize := exec.Command("powershell", []string{"-command", fmt.Sprintf("Resize-VHD %s %d", diskPath, newSize.ToBytes())}...)
```
Comment (n1hility) [inline pkg/hypervctl/disk_resize.go:15]: Note there is also an API call you can make on the ImageManagementService, similar to how the CreateVhdxFile function does, but instead against the resize method:

https://learn.microsoft.com/en-us/windows/win32/hyperv_v2/resizevirtualharddisk-msvm-imagemanagementservice

Comment (n1hility) [reply pkg/hypervctl/disk_resize.go:15]: Code would look something like this:
```go
        err = imms.BeginInvoke("ResizeVirtualHardDisk").
		In("Path", diskPath).
                In("MaxInternalSize", newSize.ToBytes()).
		Execute().
		Out("Job", &job).
		Out("ReturnValue", &ret).
		End()

	if err != nil {
		return fmt.Errorf("failed to resize disk: %w", err)
	}

	return waitVMResult(ret, service, job, "failed to resize disk", nil)
```

**Comment by baude** [comment]
once #67 merges, i will write an integration test with the current code and then update to the suggested code. that will make me feel more comfy with the change

**Comment by baude** [comment]
@n1hility I think we are good to go now ... wdyt?

@cfergeau double checking this is what you need as well...

**Comment by cfergeau** [comment]
Yes, we want to replace this:
```
	return cmd("Hyper-V\\Resize-VHD",
		"-Path",
		quote(diskPath),
		"-SizeBytes",
		fmt.Sprintf("%d", newSize))
```
The new API seems to fit.
We'd also need to be able to get the disk size for a disk image (`out, err := cmdOut(fmt.Sprintf("@(Get-VHD -Path %s).Size", quote(diskPath)))`), I don't know if it's already possible in libhvee?

**Comment by n1hility** [comment]
LGTM

**Comment by n1hility** [comment]
> Yes, we want to replace this:
> 
> ```
> 	return cmd("Hyper-V\\Resize-VHD",
> 		"-Path",
> 		quote(diskPath),
> 		"-SizeBytes",
> 		fmt.Sprintf("%d", newSize))
> ```
> 
> The new API seems to fit. We'd also need to be able to get the disk size for a disk image (`out, err := cmdOut(fmt.Sprintf("@(Get-VHD -Path %s).Size", quote(diskPath)))`), I don't know if it's already possible in libhvee?

Not yet, although there is an API call we can get it from here:
https://learn.microsoft.com/en-us/windows/win32/hyperv_v2/getvirtualharddisksettingdata-msvm-imagemanagementservice

**Comment by baude** [comment]
ill add the api now

**Comment by n1hility** [comment]
LGTM other than the go.sum conflict

File: test/e2e/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+)
+
+// PowerShellCommand is a wrapper to run something in powershell.  It takes a command and args.
+// And it returns stdout as a string
+func PowerShellCommand(commandName string, args []string) (string, error) {
```
Comment (gbraad) [inline test/e2e/utils_test.go:12]: Is this still used?

Comment (baude) [reply test/e2e/utils_test.go:12]: No, but a the test suite gets filled out, I'm certain it will be.

**Pull request merged**